src/Tizen.NUI.WindowSystem/src/public/SoftkeyService.cs

```diff
@@ -0,0 +1,292 @@
+        private Interop.SoftkeyService.SoftkeyOpacityEventCallback _onOpacityChanged;
+        
+        private event EventHandler<VisibleState> _visibleChanged;        
+        private event EventHandler<ExpandState> _expandChanged;        
+        private event EventHandler<OpacityState> _opacityChanged;
+
+        /// <summary>
+        /// Creates a new Softkey Service handle.
+        /// </summary>
+        /// <param name="tzShell">The TizenShell instance.</param>
+        /// <param name="win">The window to provide service of the quickpanel.</param>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when a argument is null.</exception>
+        public SoftkeyService(TizenShell tzShell, Window win)
+        {
+            if (tzShell == null)
+            {
+                throw new ArgumentNullException(nameof(tzShell));
+            }
+            if (tzShell.GetNativeHandle() == IntPtr.Zero)
+            {
+                throw new ArgumentException("tzShell is not initialized.");
+            }
+            if (win == null)
+            {
+                throw new ArgumentNullException(nameof(win));
+            }
+
+            _tzsh = tzShell;
+            _tzshWin = win.GetNativeId();
+            _softkeyService = Interop.SoftkeyService.Create(_tzsh.GetNativeHandle(), (IntPtr)_tzshWin);
+            if (_softkeyService == IntPtr.Zero)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _tzsh.ErrorCodeThrow(err);
+            }
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        ~SoftkeyService()
+        {
+            if (!isDisposeQueued)
+            {
+                isDisposeQueued = true;
+                DisposeQueue.Instance.Add(this);
+            }
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <exception cref="MemberAccessException">Thrown when private memeber is a corrupted.</exception>
+        public void Dispose()
+        {
+            if (isDisposeQueued)
+            {
+                Dispose(DisposeTypes.Implicit);
+            }
+            else
+            {
+                Dispose(DisposeTypes.Explicit);
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(DisposeTypes type)
+        {
+            if (!disposed)
+            {
+                if (_softkeyService != IntPtr.Zero)
+                {
+                    int res = Interop.SoftkeyService.Destroy(_softkeyService);
+                    try
+                    {
+                        _tzsh.ErrorCodeThrow(res);
```

Comment:
Don't throw an exception in Dispose methods
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1065?view=vs-2019

Reply:
I removed exception in Dispose method.